kernel_gateway/gatewayapp.py

```diff
@@ -528,5 +530,8 @@ def shutdown(self):
         """Stop all kernels in the pool."""
         self.personality.shutdown()
 
+    def _signal_stop(self, sig, frame):
+        self.log.info("Received signal to terminate.")
+        ioloop.IOLoop.current().stop()
```

Comment:
All other places use `self.io_loop`, it seems. But if that is used here, the signal handler should be installed after assigning `self.io_loop`.